airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py

```diff
@@ -139,6 +101,269 @@ def extract_schema(profiles_yml: dict) -> str:
+
+    # Generate column typing model
+    previous_sql_file_name = sql_file_name
+    table = "{{ ref('" + previous_sql_file_name + "') }}"
+    sql_file_name = normalize_identifier_name("ab2_{}".format(name), integration_type)
+    sql = "select\n  "
+    if parent_hash_id:
+        sql += "{},\n  ".format(parent_hash_id)
+    for field in properties.keys():
+        sql_field = cast_property_type(name=field, definition=properties[field], integration_type=integration_type)
+        if sql_field and not is_airbyte_column(field):
+            sql += sql_field + ",\n  "
+    sql += "_airbyte_emitted_at"
+    sql += "\nfrom {}".format(table)
+    if len(path) > 1:
+        sql += "-- {} from {}\n  ".format(name, "/".join(path))
+    output_sql_view(output, raw_schema, sql_file_name, sql, path)
+
+    hash_id = quote(f"_airbyte_{name}_hashid", integration_type)
+    # Generate hash_id column model
+    previous_sql_file_name = sql_file_name
+    table = "{{ ref('" + previous_sql_file_name + "') }}"
+    sql_file_name = normalize_identifier_name("ab3_{}".format(name), integration_type)
+    sql = "select\n  *,\n  "
+    if properties:
+        sql += "{{ dbt_utils.surrogate_key([\n    "
+        if parent_hash_id:
+            sql += "'{}',\n  ".format(parent_hash_id)
+        for field in properties.keys():
+            sql_field = safe_cast_to_varchar(name=field, definition=properties[field], integration_type=integration_type)
+            if sql_field and not is_airbyte_column(field):
+                sql += sql_field + ",\n    "
+        sql += "\n  ]) }} as " + hash_id
+    else:
+        sql += "null as " + hash_id
+    sql += "\nfrom {}".format(table)
+    if len(path) > 1:
+        sql += "-- {} from {}\n  ".format(name, "/".join(path))
+    output_sql_view(output, raw_schema, sql_file_name, sql, path)
+
+    # Generate dedup final model
+    previous_sql_file_name = sql_file_name
+    table = "{{ ref('" + previous_sql_file_name + "') }}"
+    sql_file_name = normalize_identifier_name(name, integration_type)
+    sql = """
+with numbered as (
+  select
+    *,
+    row_number() over (
+      partition by {}
+      order by _airbyte_emitted_at
+    ) as row_num
+  from {}
+)
+select
+  """.format(
+        hash_id, table
+    )
+    if parent_hash_id:
+        sql += "{},\n  ".format(parent_hash_id)
+    for field in properties.keys():
+        sql += quote(field, integration_type) + ",\n  "
+    sql += "_airbyte_emitted_at,\n  "
+    sql += hash_id
+    sql += "\nfrom numbered where row_num = 1\n"
+    if len(path) > 1:
+        sql += "-- {} from {}\n  ".format(name, "/".join(path))
+    output_sql_table(output, schema, sql_file_name, sql, path)
+
+    # Generate children models
+    previous_sql_file_name = sql_file_name
+    table = "{{ ref('" + previous_sql_file_name + "') }}"
+    for field in properties.keys():
+        if is_airbyte_column(field):
+            pass
+        elif is_combining_node(properties[field]):
```

Comment:
the behavior will be that we still will not include anyof, allof, etc in the output of normalization, right? i'm totally fine with this, but i want to make sure i understand the behavior.

Reply:
Yes, I haven't handled those cases yet because I didn't get my thoughts cleared up on what would happen if you had a combining node (`anyof`, `allof`, etc) and children nodes don't have the exact same schema...)